ompi/mca/topo/treematch/treematch/tm_tree.c

```diff
@@ -132,8 +132,8 @@ void free_non_constraint_tree(tm_tree_t *tree)
 {
   int d = tree->dumb;
 
-  free_list_child(tree);
   free_tab_child(tree);
+  free_list_child(tree);
```

Comment:
@bosilca can you please make sure this is pushed upstream ?

Reply:
Yep, I forwarded to the treematch maintainers.